### PR TITLE
Add GDPR Consent Dialog demo

### PR DESCRIPTION
**Pull request by ahmed-gz**
## Description

[YouTrack issue](https://todo.irdcorp.dev/issue/SID-000)

Added GDPR Consent Dialog demo to `react-nextjs` and `react-remix` apps to test SSR behavior.

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have generated the new version of the docs website and smoke tested it
- [ ] I have checked that my changes haven't caused semantic errors in the existing docs
- [ ] I have updated the README and DEVELOPMENT files

### CODE REVIEW

File: apps/react-nextjs/pages/gdpr-consent-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { GDPRConsentDialogPage } from "demo-form";
+
+export default function StepUpAuth() {
```
Comment (ikovic) [inline apps/react-nextjs/pages/gdpr-consent-dialog.tsx:3]: Leftover here 🔍

File: packages/demo-form/src/pages/GDPRConsentDialogPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export { GDPRConsentDialogPage } from "./gdpr-consent-dialog-page";
+
+export const GDPRConsentDialogPageRoute = "gdpr-consent-dialog";
```
Comment (ikovic) [inline packages/demo-form/src/pages/GDPRConsentDialogPage/index.tsx:3]: `"/gdpr-consent-dialog"` would be easier to use

File: apps/react-nextjs/pages/index.tsx [hunk 3]
```diff
@@ -24,6 +24,8 @@ function Home() {
           <Link href="/mfa">{`<MultiFactorAuth />`}</Link>
           <br />
           <Link href="/step-up-auth">{`<StepUpAuth />`}</Link>
+          <br />
+          <Link href="/gdpr-consent-dialog">{`<GDPRConsentDialog />`}</Link>
```
Comment (ikovic) [inline apps/react-nextjs/pages/index.tsx:28]: The idea was to import the route string from the relevant page module - looks like we haven't done this yet, but maybe we should start here by importing the route from `GDPRConsentDialogPageRoute`. cc @kaspermroz

File: apps/react-nextjs/pages/gdpr-consent-dialog.tsx (hunk 1 above)
Comment (ahmed-gz) [reply apps/react-nextjs/pages/gdpr-consent-dialog.tsx:3]: 🤦‍♂️

**Review by ikovic — APPROVED** [review]
Thank you @ahmed-gz!

**Review by kaspermroz — APPROVED** [review]
Good job @ahmed-gz ! 🚀

**Pull request merged**